kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/ElementAtEvaluator.java

```diff
@@ -0,0 +1,133 @@
+
+        if (!keyTypeFromMapInput.equivalent(lookupKeyType)) {
+            if (canCastTo(lookupKeyType, keyTypeFromMapInput)) {
+                lookupKey = new ImplicitCastExpression(lookupKey, keyTypeFromMapInput);
+            } else {
+                throw new UnsupportedOperationException(format(
+                    "%s: lookup key type (%s) is different from the map key type (%s)",
+                    elementAt, lookupKeyType, asMapType.getKeyType()));
+            }
+        }
+        return new ScalarExpression(elementAt.getName(), Arrays.asList(mapInput, lookupKey));
+    }
+
+    /**
+     * Utility method to evaluate the {@code element_at} on given map and key vectors.
+     * @param map {@link ColumnVector} of {@code map(string, string)} type.
+     * @param lookupKey {@link ColumnVector} of {@code string} type.
+     * @return
+     */
+    static ColumnVector eval(ColumnVector map, ColumnVector lookupKey) {
+        return new ColumnVector() {
+            // Store the last lookup value to avoid multiple looks up for same row id.
+            private int lastLookupRowId = -1;
+            private Object lastLookupValue = null;
+
+            @Override
+            public DataType getDataType() {
+                return ((MapType) map.getDataType()).getValueType();
+            }
+
+            @Override
+            public int getSize() {
+                return map.getSize();
+            }
+
+            @Override
+            public void close() {
+                Utils.closeCloseables(map, lookupKey);
+            }
+
+            @Override
+            public boolean isNullAt(int rowId) {
+                if (rowId == lastLookupRowId) {
+                    return lastLookupValue == null;
+                }
+                return map.isNullAt(rowId) || lookupValue(rowId) == null;
+            }
+
+            @Override
+            public String getString(int rowId) {
+                lookupValue(rowId);
+                return lastLookupValue == null ? null : (String) lastLookupValue;
+            }
+
+            private Object lookupValue(int rowId) {
+                if (rowId == lastLookupRowId) {
+                    return lastLookupValue;
+                }
+                // TODO: this needs to be updated after the new way of accessing the complex
+                //  types is merged.
+                lastLookupRowId = rowId;
+                String keyValue = lookupKey.getString(rowId);
+                lastLookupValue = map.getMap(rowId).get(keyValue);
+                return lastLookupValue;
+            }
+        };
+    }
+
+    private static MapType validateSupportedMapType(Expression elementAt, DataType mapInputType) {
+        checkArgument(
+            mapInputType instanceof MapType,
+            "expected a map type input as first argument: " + elementAt);
+        MapType asMapType = (MapType) mapInputType;
+        // TODO: we may extend type support in future, but for the need is just a look
```

Comment:
nit: incomplete sentence `but for the need is just a look`? maybe mission a word or a wrong word